### PR TITLE
use the repo_tool as part of the CI

### PR DESCRIPTION
**Pull request by devoncarew**
- use the repo_tool as part of the CI
- introduce a `repo_tool repo-check` command w/ the contents of the previous `packages/devtools_app/tool/version_check.dart` file
- fix an issue with recursing into the `//flutter/` directory on the CI

### CODE REVIEW

**Comment by devoncarew** [comment]
Still tweaking this, but it looks like the goldens might need to be updated.

**Comment by devoncarew** [comment]
Merged w/ master to get the test goldens update.

File: tool/lib/commands/repo_check.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  @override
+  String get name => 'devtools version';
+
+  @override
+  Future<void> performCheck(DevToolsRepo repo) {
+    final pubspecContents = repo.readFile('packages/devtools_app/pubspec.yaml');
+    final versionString = pubspecContents
+        .split('\n')
+        .firstWhere((line) => line.startsWith('version:'));
+    final pubspecVersion = versionString.substring('version:'.length).trim();
```
Comment (jacob314) [inline tool/lib/commands/repo_check.dart]: nit: add a TODO to use a package to parse pubspec files
https://pub.dev/packages/pubspec

Comment (devoncarew) [reply tool/lib/commands/repo_check.dart]: will do

**Pull request merged**